src/background/services/monetization.ts

```diff
@@ -317,9 +353,13 @@ export class MonetizationService {
     const isSiteMonetized = tab?.id ? this.sessions[tab.id]?.size > 0 : false
 
     return {
-      ...storedData,
+      ...dataFromStorage,
       balance: balance.total.toString(),
       url,
+      grants: {
+        oneTime: oneTimeGrant?.amount,
+        recurring: recurringGrant?.amount
+      },
```

Comment:
Why do we need to send the amounts as well? Aren't they computed within `this.storage.getBalance` ?

Reply:
I'm passing the amount to UI so we can show a default value in input where user will add funds, and make it possible also to suggest amounts.
The recurring one also includes interval, which we use to show when will the money be added automatically for use, if the user chooses to do nothing.
I'm passing raw inputs so we can compute those details as and when needed in the popup. I'm passing just the amounts as that's what we need in popup, and avoiding passing tokens to reduce security risks.